AssetRipper.Library/Exporters/Shaders/UnityPatch/YamlShaderPostprocessor.txt

```diff
@@ -0,0 +1,25 @@
+using UnityEngine;
+using UnityEditor;
+
+namespace AssetRipperPatches.Editor
+{
+    /// <summary>
+    /// This script is AssetRipper's patch for shaders exported as YAML assets.
+    /// Such a shader can be assigned to and used by a material as a regular .shader asset,
+    /// but it does not work with Shader.Find() unless we explicitly register it. 
+    /// </summary>
+    public class AssetRipperShaderPostprocessor : AssetPostprocessor
+    {
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
+        {
+            foreach (var importedAsset in importedAssets)
+            {
+                if (!importedAsset.EndsWith(".asset")) continue;
+                Shader yamlShader = AssetDatabase.LoadMainAssetAtPath(importedAsset) as Shader;
+                if (yamlShader == null) continue;
+                ShaderUtil.RegisterShader(yamlShader);
```

Comment:
Is this function forgiving?
* What happens if this is used multiple times on the same shader?
* What about a normal shader? If you didn't have a file extension check, would that break things?

Reply:
What happens if this is used multiple times on the same shader?
Totally allowed. Same effect as a single time.

What about a normal shader? If you didn't have a file extension check, would that break things?
A normal shader is automatically registered. Registering it again by using ShaderUtil.RegisterShader is much the same as the situation in the first question - registering multiple times.
The file extension check is solely for lowering the performance cost, as we don't want to load all meshes and textures in the project.

Reply:
Perfect